src/repr/src/proto.rs

```diff
@@ -251,6 +261,10 @@ impl<T: ProtoRepr> ProtoRepr for Option<T> {
     }
 }
 
+pub fn any_duration() -> impl Strategy<Value = std::time::Duration> {
+    (0..u64::MAX, 0..u32::MAX).prop_map(|(secs, nanos)| std::time::Duration::new(secs, nanos))
```

Comment:
the valid range for nanos is only (0 .. NANOS_PER_SEC) , excluding.

`const NANOS_PER_SEC: u32 = 1_000_000_000;` is unfortunately not public, but copy paste will work ;-)

Reply:
this is my only needs fix for approval comment

Reply:
Ah, nice catch, I didn't know about this! Interesting that the tests didn't catch it.

Reply:
(it does make a lot of sense to ensure that durations have a single unique representation, though)

Reply:
Fixed in a779958bb